framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2061,6 +2061,75 @@ protected void adaptGetClassReturnTypeToReceiver(
         classWildcardArg.setExtendsBound(newBound);
     }
 
+    /**
+     * Return the element type of the given iterator. This is usually the type of {@code
+     * expression.itertor().next()}. If {@code expression} is an array, it is the component type of
+     * the array.
+     *
+     * @param expression an expression whose type is an array or implements {@link Iterable}
+     * @return the type of {@code expression.itertor().next()} or if {@code expression} is an array,
+     *     the component type of the array.
+     */
+    public AnnotatedTypeMirror getIteratedType(ExpressionTree expression) {
```

Comment:
I realize the terminology existed before this pull request, but I find "iterated type" confusing because it's new, nonstandard jargon.  Consider renaming the methods, such as to `getIteratorElementType`.